packages/itwin/property-grid/src/components/PropertyGrid.tsx

```diff
@@ -160,6 +162,21 @@ export const PropertyGrid = ({
     };
   }, []);
 
+  // If persisting hide/show empty values, get the preference
+  useMemo(() => {
```

Comment:
useMemo if for setting a value that needs to be changed when the deps change, useCallback is for setting a function, and useEffect is for running some code, which in this case is the right one to use